src/dashboard/src/components/administration/views.py

```diff
@@ -223,6 +229,160 @@ def storage(request):
+                dirs[directory]['used'] = _usage_check_directory_used(dirs[directory]['path'])
+
+    return dirs
+
+def _usage_check_directory_volume_size(path):
+    # Get volume size (in 512 byte blocks)
+    try:
+        output = subprocess.check_output(["df", path])
+        device, size, used, available, percent, mountpoint = output.split("\n")[1].split()
+        return int(size) * 512
+    except subprocess.CalledProcessError:
+        return 0
+
+def _usage_check_directory_used(path):
+    # Get total usage in bytes
+    try:
+        output = subprocess.check_output(["du", "--bytes", "--summarize", path])
+        return output.split("\t")[0]
+    except subprocess.CalledProcessError:
+        return 0
+
+def _usage_shared_dir_path():
+    # Get shared directory path
+    clientConfigFilePath = os.path.join(os.sep, 'etc', 'archivematica', 'MCPClient', 'clientConfig.conf')
+    config = ConfigParser.SafeConfigParser()
+    config.read(clientConfigFilePath)
+
+    return config.get('MCPClient', 'sharedDirectoryMounted')
+
+def _usage_location_path(purpose):
+    # Get currently processing location
+    locations = storage_service.get_location(purpose=purpose)
+    location = locations[0]
+    return location['path']
+
+def clear_context(request, dir_id):
+    usage_dirs = _usage_dirs(False)
+    prompt = 'Clear ' + usage_dirs[dir_id]['description'] + '?'
+    cancel_url = reverse("components.administration.views.usage")
+    return RequestContext(request, {'action': 'Delete', 'prompt': prompt, 'cancel_url': cancel_url})
+
+@user_passes_test(lambda u: u.is_superuser, login_url='/forbidden/')
+@decorators.confirm_required('simple_confirm.html', clear_context)
+def usage_clear(request, dir_id):
+    if request.method == 'POST':
+        usage_dirs = _usage_dirs(False)
+        dir_info = usage_dirs[dir_id]
+
+        # Prevent shared directory from being cleared
+        if dir_id == 'shared' or not dir_info:
+            raise Http404
+
+        # Determine if specific subdirectories need to be cleared, rather than
+        # whole directory
+        if dir_info.get('subdirectories') is not None:
+            dirs_to_empty = []
+            for subdirectory in dir_info['subdirectories']:
+                dirs_to_empty.append(os.path.join(dir_info['path'], subdirectory))
+        else:
+            dirs_to_empty = [dir_info['path']]
+
+        # Attempt to clear directories
+        successes = []
+        errors = []
+
+        for directory in dirs_to_empty:
+            try:
+                for entry in os.listdir(directory):
+                    entry_path = os.path.join(directory, entry)
+                    if os.path.isfile(entry_path):
+                        os.unlink(entry_path)
+                    else:
+                        shutil.rmtree(entry_path)
+                successes.append(directory)
+            except Exception, e:
+                errors.append(str(e))
```

Comment:
Do we want to log these as well?  logger.exception or logger.foo(..., exc_info=True) may be useful.